plugin/storage/memory/memory.go

```diff
@@ -121,6 +125,43 @@ func (st *Store) GetDependencies(ctx context.Context, endTs time.Time, lookback
 					} else {
 						deps[depKey].CallCount++
 					}
+				} else if isClientSpan(s) && !hasCorrespondingServerSpan(s, trace.Spans) {
+
+					// Generate a unique ID based on current time and a counter
+					currentTime := uint64(time.Now().UnixNano())
+					uniqueID := (currentTime << 32) | (atomic.AddUint64(&counter, 1) & 0xFFFFFFFF)
+
+					// Creating an artificial server span
+					artificialSpan := &model.Span{
```

Comment:
why do you need to create artificial span? All this method needs to do is to return dependency links, which you already construct in L157

Reply:
I've updated the code to focus on updating dependency links directly, eliminating the need for artificial spans.